dask/dataframe/core.py

```diff
@@ -5088,6 +5093,41 @@ def applymap(self, func, meta="__no_default__"):
     def round(self, decimals=0):
         return elemwise(M.round, self, decimals)
 
+    @derived_from(pd.DataFrame)
+    def nunique(self, split_every=False, dropna=True, axis=0):
+        if axis == 1:
+            # split_every not used for axis=1
+            meta = self._meta_nonempty.nunique(axis=axis)
+            return self.map_partitions(
+                M.nunique,
+                meta=meta,
+                token="series-nunique",
+                axis=axis,
+                dropna=dropna,
+                enforce_metadata=False,
+            )
+        else:
+            nunique_list = []
+            for col in self.columns:
+                nunique = Series.nunique(
+                    self[col], split_every=split_every, dropna=dropna
+                )
+                nunique.name = self[col].name
+                nunique_list.append(nunique)
```

Comment:
This feels cleaner to me, am I missing something?

```suggestion
            nunique_list = [
                self[col].nunique(split_every=split_every, dropna=dropna) 
                for col in self.columns
            ]
```

Reply:
Good point, it looks equivalent to me as well. I suspect this pattern was copied over from `DataFrame.mode`, though I don't know why it's written that way.

Reply:
thanks, just fixed this! and yup, I had it like this since I copied it from `DataFrame.mode`